components/service/glean/src/main/java/mozilla/components/service/glean/utils/TestUtils.kt

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.utils
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import mozilla.components.service.glean.Glean
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.scheduler.PingUploadWorker
+import mozilla.components.service.glean.storages.ExperimentsStorageEngine
+import mozilla.components.service.glean.storages.StorageEngineManager
+import java.util.concurrent.ExecutionException
+
+object TestUtils {
+    /**
+     * Resets the Glean state and trigger init again.
+     *WorkManagerTestInitHelper
+     * @param context the application context to init Glean with
+     * @param config the [Configuration] to init Glean with
+     * @param clearStores if true, clear the contents of all stores
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
```

Comment:
I'd rather not increase our API surface, if possible. Looks like your use-case could be satisfied by slightly refactoring the code, as described above. I'm happy to have a meeting to further discuss this! I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1556985) for providing an example. Once again, thank you very much for helping us on this.

Reply:
I think this function in particular might need to be public-for-testing though.   Being able to clear the stores between tests feels pretty required...

Reply:
In the latest patch, I factored out the logic to reset Glean into a test-only `Glean.reset` function, and changed `resetGlean` in the Glean and Sync tests to call it after enabling testing and initializing the test work manager.